tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs

```diff
@@ -0,0 +1,122 @@
+namespace VstsPipelineSync
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data.SqlClient;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using DevOpsLib;
+    using DevOpsLib.VstsModels;
+
+    public class VstsBuildBatchUpdate
+    {
+        readonly DevOpsAccessSetting devOpsAccessSetting;
+        readonly string dbConnectionString;
+
+        public VstsBuildBatchUpdate(DevOpsAccessSetting devOpsAccessSetting, string dbConnectionString)
+        {
+            ValidationUtil.ThrowIfNull(devOpsAccessSetting, nameof(devOpsAccessSetting));
+
+            this.devOpsAccessSetting = devOpsAccessSetting;
+            this.dbConnectionString = dbConnectionString;
+        }
+
+        public async Task RunAsync(TimeSpan waitPeriodAfterEachUpdate, CancellationToken ct)
+        {
+            var buildManagement = new BuildManagement(devOpsAccessSetting);
+            var lastUpdatePerBranchPerDefinition = new Dictionary<string, Dictionary<BuildDefinitionId, DateTime>>();
+
+            while (!ct.IsCancellationRequested)
+            {
+                try
+                {
+                    Console.WriteLine($"Import Vsts Builds data started at {DateTime.UtcNow}");
+
+                    lastUpdatePerBranchPerDefinition.Upsert("refs/heads/master", await ImportVstsBuildsDataAsync(buildManagement, "refs/heads/master", BuildExtension.MasterBranchBuildDefinitions,  lastUpdatePerBranchPerDefinition));
+                }
```

Comment:
nit: I found this confusing when I first read it. Might make it easier to split into two lines.